lib/api/bucketPutLifecycle.js

```diff
@@ -0,0 +1,68 @@
+const { waterfall } = require('async');
+const LifecycleConfiguration =
+    require('arsenal').models.LifecycleConfiguration;
+
+const parseXML = require('../utilities/parseXML');
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+const metadata = require('../metadata/wrapper');
+const { metadataValidateBucket } = require('../metadata/metadataUtils');
+const { pushMetric } = require('../utapi/utilities');
+
+/**
+ * Bucket Put Versioning - Create or update bucket lifecycle configuration
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+
+function bucketPutLifecycle(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'bucketPutLifecycle' });
+
+    const bucketName = request.bucketName;
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        requestType: 'bucketOwnerAction',
+    };
+    return waterfall([
+        next => parseXML(request.post, log, next),
+        (parsedXml, next) => {
+            const lcConfigClass = new LifecycleConfiguration(parsedXml);
+            // if there was an error getting lifecycle configuration,
+            // returned configObj will contain 'error' key
+            process.nextTick(() => {
+                const configObj = lcConfigClass.getLifecycleConfiguration();
+                return next(configObj.error || null, configObj);
+            });
+        },
+        (lcConfig, next) => metadataValidateBucket(metadataValParams, log,
+            (err, bucket) => {
+                if (err) {
+                    return next(err, bucket);
```

Comment:
No need for second arg to `next` here for error condition.

Reply:
Oh wait, I guess this is for collecting CORS headers at the bottom?

Reply:
yup, for the corsHeaders